plugins/modules/github_webhook.py

```diff
@@ -154,13 +154,18 @@
 
 
 def _create_hook_config(module):
-    return {
+    hook_config = {
         "url": module.params["url"],
         "content_type": module.params["content_type"],
-        "secret": module.params.get("secret"),
         "insecure_ssl": "1" if module.params["insecure_ssl"] else "0"
     }
 
+    secret = module.params.get("secret")
+    if secret is not None and len(secret) > 0:
```

Comment:
You can also shorten this to:
```suggestion
    if secret:
```